packages/dx-react-scheduler/src/plugins/day-view.tsx

```diff
@@ -28,26 +30,57 @@ const endViewDateBaseComputed = ({
   viewCellsData,
 }) => endViewDateCore(viewCellsData);
 
-export class DayView extends React.PureComponent {
+class DayViewBase extends React.PureComponent<VerticalViewProps, ViewState> {
+  static defaultProps = {
+    name: 'Day',
+    startDayHour: 0,
+    endDayHour: 24,
+    cellDuration: 30,
+    intervalCount: 1,
+  };
+
+  static components = {
+    layoutComponent: 'Layout',
+    appointmentLayerComponent: 'AppointmentLayer',
+    dayScaleEmptyCellComponent: 'DayScaleEmptyCell',
+    timeScaleLayoutComponent: 'TimeScaleLayout',
+    timeScaleCellComponent: 'TimeScaleCell',
+    timeScaleRowComponent: 'TimeScaleRow',
+    dayScaleLayoutComponent: 'DayScaleLayout',
+    dayScaleCellComponent: 'DayScaleCell',
+    dayScaleRowComponent: 'DayScaleRow',
+    timeTableLayoutComponent: 'TimeTableLayout',
+    timeTableCellComponent: 'TimeTableCell',
+    timeTableRowComponent: 'TimeTableRow',
+  };
+
+  startViewDateComputed: (getters: Getters) => any;
+  endViewDateComputed: (getters: Getters) => any;
+  availableViewNamesComputed: (getters: Getters) => any;
+  currentViewComputed: (getters: Getters) => any;
+  intervalCountComputed: (getters: Getters) => any;
+  cellDurationComputed: (getters: Getters) => any;
+  viewCellsDataComputed: (getters: Getters) => any;
+  timeTable: any;
```

Comment:
Some of this properties can be instantiated here. Also we should avoid 'any' type if possible

Reply:
Should we make many private types? We support TypeScript for users and these types are only for us and our problems.

Reply:
In general, the more restrictive types we have, the less errors we make so in perspective, adding private types is better than using 'any'

Reply:
For users only it would be simpler to manually generate *d.ts* files (like */api/\*/api.ts* files that we have now). Our code would have not be noised with ts definitions then.

Since we have ts in our code we should keep it in order.

----
1. `(getters: Getters) => any` - this function type should be extracted to a single definition (because it is duplicated several times in this files and in some other files).
2. This type is actually from *dx-react-core* package (hence should be imported, not declared here) - it is type of `computed` property of `Getter` component.